cmd/cli/app/provider/provider.go

```diff
@@ -37,9 +36,9 @@ var ProviderCmd = &cobra.Command{
 func init() {
 	app.RootCmd.AddCommand(ProviderCmd)
 	// Flags for all subcommands
-	// TODO: remove the provider flag from here and add it only to the subcommands that need it
-	ProviderCmd.PersistentFlags().StringP("provider", "p", ghclient.Github, "Name of the provider, i.e. github")
 	ProviderCmd.PersistentFlags().StringP("project", "j", "", "ID of the project")
+	// TODO: get rid of this
+	ProviderCmd.PersistentFlags().StringP("provider", "p", "", "DEPRECATED - use `class` flag of `enroll` instead")
```

Comment:
Since this was only useful for the `provider enroll` command, should it be moved there? Otherwise other commands like `provider list` see the deprecation message

Reply:
I left it that way for the sake of backwards compatibility. I also think that we should consider changing the name of the parameter: in some commands elsewhere in the CLI, `provider` might mean the name of the provider, here it means the class/type